### PR TITLE
ChatInfoPage as attached page

### LINKED ISSUE #150: Access Chat information page with a swipe
Instead to access the chat information page only by clicking on the header it would be nice also with a swipe from right to left

### LINKED ISSUE #166: Group members list is repeated several times for some groups
The group member list in the chat info page (opened by taping the top bar in any group chat) seems to be repeated several times for some groups, eg. the Fernschreiber Development group. I checked several groups and most of them had this bug, but didn't find any clear indicators which groups would or would not trigger this.

### PR DESCRIPTION
**Pull request by jgibbon**
fixes #150: Now basically everything is inside a loader; ChatInformationPage is added to ChatPage with pageStack.pushAttached

fixes #166: Replaces the clunky VisualItemModel in tab view and doesn't initialize multiple times.

### CODE REVIEW

**Comment by jgibbon** [comment]
Whoops, accidentally rebased to monichs nomutex branch, but shouldn't interfere with merging.

File: qml/pages/ChatPage.qml [hunk 1]
```diff
@@ -298,6 +298,8 @@ Page {
         case PageStatus.Active:
             if (!chatPage.isInitialized) {
                 chatModel.initialize(chatInformation);
+
+                pageStack.pushAttached(Qt.resolvedUrl("../pages/ChatInformationPage.qml"), { "chatInformation" : chatInformation, "privateChatUserInformation": chatPartnerInformation, "groupInformation": chatGroupInformation, "chatOnlineMemberCount": chatOnlineMemberCount});
```
Comment (monich) [inline qml/pages/ChatPage.qml:302]: Why "`../pages/"`?

File: qml/components/chatInformationPage/ChatInformationPageContent.qml [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+import "../js/twemoji.js" as Emoji
+import "../js/functions.js" as Functions
+
+
+SilicaFlickable {
+    id: pageContent
+    property alias membersList: membersList
+
+    function initializePage() {
+        console.log("[ChatInformationPage] Initializing chat info page...");
```
Comment (monich) [inline qml/components/chatInformationPage/ChatInformationPageContent.qml:31]: I suggest to not leave console logs in files committed to master

File: qml/components/chatInformationPage/ChatInformationPageContent.qml [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+        leftMargin: imageContainer.minDimension + Theme.horizontalPageMargin + Theme.paddingMedium
+        title: chatInformationPage.chatInformation.title !== "" ? Emoji.emojify(chatInformationPage.chatInformation.title, Theme.fontSizeLarge, emojiBase) : qsTr("Unknown")
+        description: chatInformationPage.isPrivateChat ? ("@"+(chatInformationPage.privateChatUserInformation.username || chatInformationPage.chatPartnerGroupId)) : ""
+    }
+
+    SilicaFlickable {
+        id: contentFlickable
+        contentHeight: groupInfoItem.height + tabViewLoader.height
+        clip: true
+        interactive: true//groupInfoItem.height > pageContent.height * 0.5
```
Comment (monich) [inline qml/components/chatInformationPage/ChatInformationPageContent.qml:297]: `interactive` should be false when scroll animation is running, e.g.
```qml
interactive: !scrollUpAnimation.running && !scrollDownAnimation.running
```

File: qml/components/chatInformationPage/ChatInformationPageContent.qml [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+                text: (chatInformationPage.isPrivateChat && chatInformationPage.privateChatUserInformation.phone_number ? "+"+chatInformationPage.privateChatUserInformation.phone_number : "") || ""
+                isLinkedLabel: true
+            }
+
+            SectionHeader {
+                font.pixelSize: Theme.fontSizeExtraSmall
+                visible: !!inviteLinkItem.text
+                height: visible ? Theme.itemSizeExtraSmall : 0
+                text: qsTr("Invite Link", "header")
+                x: 0
```
Comment (monich) [inline qml/components/chatInformationPage/ChatInformationPageContent.qml:405]: `x` is zero by default

File: qml/components/chatInformationPage/ChatInformationTabItemBase.qml [hunk 5]
```diff
@@ -30,15 +30,12 @@ Item {
...
+    property bool active: index === tabView.currentIndex
 
     default property alias _data: contentItem.data
 
-
-    width: parent.width
-    height: tabView.maxHeight//Math.max(contentItem.height, loadingColumn.height)
-    opacity: active ? 1.0 : 0.0
+    opacity: active ? 1.0 : 0.2
     Behavior on opacity { PropertyAnimation {duration: 300}}
```
Comment (monich) [inline qml/components/chatInformationPage/ChatInformationTabItemBase.qml:39]: Better to use `FadeAnimation` for all opacity behaviors for visual consistency. Unless this one is made 100ms longer than `FadeAnimation` on purpose (which I doubt)

**Comment by jgibbon** [comment]
Wow, thanks for taking the time to have much more than a look at this. I agree with all of the points raised and will address them ASAP.

File: qml/components/chatInformationPage/ChatInformationPageContent.qml (hunk 4 above)
Comment (jgibbon) [reply qml/components/chatInformationPage/ChatInformationPageContent.qml:405]: For SectionHeader, the default is `x: Theme.horizontalPageMargin` AFAICT

File: qml/pages/ChatPage.qml (hunk 1 above)
Comment (jgibbon) [reply qml/pages/ChatPage.qml:302]: Good point, will change.

File: qml/components/chatInformationPage/ChatInformationPageContent.qml (hunk 2 above)
Comment (jgibbon) [reply qml/components/chatInformationPage/ChatInformationPageContent.qml:31]: Perhaps we should think about log output in general (also c++) in the future. 
Only enable in debug builds? Command line parameter? 
But that's a topic for another place. Will remove here.

File: qml/components/chatInformationPage/ChatInformationPageContent.qml (hunk 3 above)
Comment (jgibbon) [reply qml/components/chatInformationPage/ChatInformationPageContent.qml:297]: Good catch!

File: qml/components/chatInformationPage/ChatInformationTabItemBase.qml (hunk 5 above)
Comment (jgibbon) [reply qml/components/chatInformationPage/ChatInformationTabItemBase.qml:39]: I actually like 300ms, but you're right. 
Also I missed the 0.2 above which I didn't want to commit.

**Comment by jgibbon** [comment]
I've adressed the mentioned issues (plus a few more) and rebased to current master.

File: qml/components/chatInformationPage/ChatInformationPageContent.qml (hunk 4 above)
Comment (monich) [reply qml/components/chatInformationPage/ChatInformationPageContent.qml:404]: Ah, you're getting rid of margins. OK.

File: qml/components/chatInformationPage/ChatInformationPageContent.qml (hunk 2 above)
Comment (monich) [reply qml/components/chatInformationPage/ChatInformationPageContent.qml]: That's one of the reasons why I'm turning those into macros on C++ size - to make it easier to manage those, even strip them completely at compile time.

File: qml/components/chatInformationPage/ChatInformationTabItemBase.qml (hunk 5 above)
Comment (monich) [reply qml/components/chatInformationPage/ChatInformationTabItemBase.qml]: It looks disturbing when different things on the same screen are being animated slightly out of sync))

**Comment by Wunderfitz** [comment]
Works great in most cases. However, a small thing: Please check out channels - it tries to load members of them which results in a neverending spinning wheel. Please try to restore the old behaviour for channels. :)

**Comment by jgibbon** [comment]
@Wunderfitz I think this may be ready now. 
I don't know why github shows conflicts here – my changes should be rebased on master/your merge already.
 
Regarding #170 I've left the header image bigger compared to the ChatPage for now, but the alignment is changed a bit – Silica PageHeader centers the main title to the PageStackIndicator and generally uses a bit more space vertically.

Also, more space is used horizontally for text now when the image is "transitioned down". 
This may be useful for long chat names.

**Comment by monich** [comment]
> Regarding #170 I've left the header image bigger compared to the ChatPage for now, but the alignment is changed a bit – Silica PageHeader centers the main title to the PageStackIndicator and generally uses a bit more space vertically.

And in case if you haven't noticed, `PageHeader` has different height in landscape and portrait.

**Comment by Wunderfitz** [comment]
Yep, looks really good now. If there are other tweaks (e.g. with the page header), we can easily integrate them later. I will merge this one now. Thank you very much!

**Comment by jgibbon** [comment]
> And in case if you haven't noticed, `PageHeader` has different height in landscape and portrait.

That was handled in the initial implementation
`property int minDimension: chatInformationPage.isLandscape ? Theme.itemSizeSmall : Theme.itemSizeMedium`

But you are right, it's an easy thing to overlook, so better be safe than sorry. :)

**Pull request merged**